j5/backends/base.py

```diff
@@ -1,10 +1,44 @@
 """The base classes for backends."""
 
-from abc import ABCMeta, abstractmethod
+from abc import ABCMeta
 from typing import Any, List, Mapping
 
 
-class Backend(metaclass=ABCMeta):
+class BackendMeta(ABCMeta):
+    """
+    The metaclass for a backend.
+
+    Responsible for registering the backend with a BackendGroup.
+
+    """
+
+    def __new__(mcs, name, bases, namespace, **kwargs):
+        """Create a new class object."""
+        cls = super().__new__(mcs, name, bases, namespace, **kwargs)  # type: ignore
+
+        if cls.__name__ == "Backend":
```

Comment:
You should be able to do `if isinstance(cls, Backend)` here, rather than relying on the class name directly

Reply:
That wouldn't work? `Backend` is defined after `BackendMeta`, and is thus not in scope?

Reply:
Although I could just be being tired.